ludwig/modules/embedding_modules.py

```diff
@@ -81,33 +81,16 @@ def embedding_matrix(
             embedding_initializer_obj = embedding_initializer_obj_ref(
                 [vocab_size, embedding_size])
 
-        '''
-        embeddings = tf.Variable(
-            embedding_initializer_obj,
-            trainable=embeddings_trainable,
-            name='embeddings'
-        )
-        '''
         embeddings = embedding_initializer_obj
 
     elif representation == 'sparse':
         embedding_size = vocab_size
-        '''
-        embeddings = tf.Variable(
-            get_initializer('identity')([vocab_size, embedding_size]),
-            trainable=False,
-            name='embeddings'
-        )
-        '''
         embeddings = torch.tensor(
             get_initializer('identity')([vocab_size, embedding_size]),
```

Comment:
there's an error here that probably was there since porting to TF2. the tesnor should be `[vocab_size, vocab_size]`, otherwise every sparse representation for tokens with id > vocab_size will be the same (all 0s)

Reply:
@w4nderlust in line 94 `embedding_size` is set to be equal to `vocab_size`, so the equivalent sizes should be the same.
I can make the change regardless tho if you'd prefer

Reply:
My bad, missed it.